core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -317,6 +322,77 @@ public String toString() {
   }
 
   /**
+   * Provides arguments for a {@link SubchannelPicker#pickSubchannel(PickSubchannelArgs)}.
+   */
+  public abstract static class PickSubchannelArgs {
```

Comment:
I guess the only change needed would be to either add interface in `LoadBalancer2` or promote `PickSubchannelArgs` from internal to `io.grpc`. Probably latter, since this will let implementors re-use it in tests.